### PR TITLE
Add unit tests for the Google Books providers

### PR DESCRIPTION
**Pull request by Pithaya**
Adds unit tests for the Google Books providers and fixes some issues:

- The comic file name format from the [docs](https://jellyfin.org/docs/general/server/media/comics/) (`Series_Name (SeriesYear) #IssueNum (of Count) (PubYear).ext`) was not being matched
- `GetSearchResults` was not setting the provider id, so picking a search result wasn't working (see: https://github.com/jellyfin/jellyfin/issues/5437)
- Genres / tags were not parsed correctly
- The PersonInfo metadata was not parsed

I used the same test project as #70 to minimize conflicts.

(By the way, it would be nice if this repo could also get the hacktoberfest topic 🙂)

### CODE REVIEW

File: Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksImageProvider.cs [hunk 1]
```diff
@@ -84,9 +84,7 @@ public async Task<IEnumerable<RemoteImageInfo>> GetImages(BaseItem item, Cancell
 
             using var response = await httpClient.GetAsync(url, cancellationToken).ConfigureAwait(false);
 
-            #pragma warning disable CA2007
-            await using var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
-            #pragma warning restore CA2007
+            using var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
```
Comment (crobibero) [inline Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksImageProvider.cs:87]: I would revert this change

File: Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksProvider.cs [hunk 2]
```diff
@@ -221,9 +252,7 @@ public async Task<HttpResponseMessage> GetImageResponse(string url, Cancellation
 
             using var response = await httpClient.GetAsync(url, cancellationToken).ConfigureAwait(false);
 
-            #pragma warning disable CA2007
-            await using var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
-            #pragma warning restore CA2007
+            using var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
```
Comment (crobibero) [inline Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksProvider.cs:255]: revert

File: Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksImageProvider.cs (hunk 1 above)
Comment (Pithaya) [reply Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksImageProvider.cs:87]: I'm wondering if maybe the `stream` assignment could be removed entirely by replacing `response.Content.ReadAsStreamAsync` with `response.Content.ReadFromJsonAsync`, to make it consistent with how it's done in the main Jellyfin repo ?

Comment (crobibero) [reply Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksImageProvider.cs:87]: I agree with this suggestion

**Review by crobibero — APPROVED** [review]
